src/betanc.jl

```diff
@@ -0,0 +1,165 @@
+const errmax = 1e-15
+
+#Compute tail of noncentral Beta distribution
+#Russell Lenth, Algorithm AS 226: Computing Noncentral Beta Probabilities,
+#Applied Statistics,Volume 36, Number 2, 1987, pages 241-244
+
+"""
+	ncbeta_tail(x,a,b,lambda)
+
+Compute tail of the noncentral beta distribution.
+Uses the recursive relation 
+```math
+I_{x}(a,b+1;0) = I_{x}(a,b;0) - \\Gamma(a+b)/\\Gamma(a+1)\\Gamma(b)x^{a}(1-x)^{b}
+```
+and ``\\Gamma(a+1) = a\\Gamma(a)`` given in Equation (26.5.16) in Abromowitz and Stegun(1965).
+"""
+function ncbeta_tail(a::Float64, b::Float64, lambda::Float64, x::Float64)
+    if x <= 0.0
+        return 0.0
+    elseif x >= 1.0
+        return 1.0
+    end
+    
+    c = 0.5*lambda
+    #Init series
+
+    beta = logabsbeta(a,b)[1]
+    temp = beta_inc(a,b,x)[1]
+    gx = (beta_integrand(a,b,x,1.0-x))/a
+    q = exp(-c)
+    xj = 0.0
+    ax = q*temp
+    sumq = 1.0 - q
+    ans = ax
+
+    while true
+        xj += 1.0
+        temp -= gx
+        gx *= x*(a+b+xj-1.0)/(a+xj)
+        q *= c/xj
+        sumq -= q
+        ax = temp*q
+        ans += ax
+
+        #Check convergence
+        errbd = abs((temp-gx)*sumq)
+        if xj > 1000 || errbd < 1e-10
+            break
+        end
+    end
+    return ans
+end
+
+#R Chattamvelli, R Shanmugam, Algorithm AS 310: Computing the Non-central Beta Distribution Function,
+#Applied Statistics, Volume 46, Number 1, 1997, pages 146-156
+
+"""
+	ncbeta(a,b,lambda,x)
+
+Compute the CDF of the noncentral beta distribution given by
+```math
+I_{x}(a,b+1;\\lambda ) = \\sum_{j=0}^{\\infty}q(\\lambda/2,j)I_{x}(a+j,b;0)
+```
+For lambda < 54 : algorithm suggested by Lenth(1987) in ncbeta_tail(x,a,b,lambda).
+Else for lambda >= 54 :
+First ``\\lambda/2`` is calculated and the Poisson term is calculated using ``P(j-1)=j/\\lambda P(j)`` and ``P(j+1) = \\lambda/(j+1) P(j)``.
+Then backward recurrences are used until either the Poisson weights fall below `errmax` or `iterlo` is reached.
+```math
+I_{x}(a+j-1,b) = I_{x}(a+j,b) + \\Gamma(a+b+j-1)/\\Gamma(a+j)\\Gamma(b)x^{a+j-1}(1-x)^{b}
+```
+Then forward recurrences are used until error bound falls below `errmax`.
+```math
+I_{x}(a+j+1,b) = I_{x}(a+j,b) - \\Gamma(a+b+j)/\\Gamma(a+j)\\Gamma(b)x^{a+j}(1-x)^{b}
+```
+"""
+function ncbeta(a::Float64, b::Float64, lambda::Float64, x::Float64)
+    ans = x
+    if x <= 0.0
+        return 0.0
+    elseif x >= 1.0
+        return 1.0
+    end
+    c = 0.5*lambda
+    xj = 0.0
+
+    if lambda < 54.0
+        return ncbeta_tail(a,b,lambda,x)
+    else
+        m = round(Int, c)
+        mr = float(m)
+        iterlo = m - trunc(Int, 5.0*sqrt(mr))
+        iterhi = m + trunc(Int, 5.0*sqrt(mr))
+        t = -c + mr*log(c) - logabsgamma(mr + 1.0)[1]
+        q = exp(t)
+        r = q
+        psum = q
+
+        beta = logabsbeta(a+mr,b)[1]
+        gx = SpecialFunctions.beta_integrand(a+mr,b,x,1.0-x)/(a + mr)

```

Comment:
`SpecialFunctions` shouldn't be necessary here.

Reply:
Sorry! Forgot to remove this one.